skyportal/handlers/api/followup_request.py

```diff
@@ -288,19 +353,48 @@ def post(self):
         """
         data = self.get_json()
         _ = Source.get_obj_if_owned_by(data["obj_id"], self.current_user)
-        data["start_date"] = arrow.get(data["start_date"]).datetime
-        data["end_date"] = arrow.get(data["end_date"]).datetime
         data["requester_id"] = self.associated_user_object.id
-        if isinstance(data["filters"], str):
-            data["filters"] = [data["filters"]]
-        followup_request = FollowupRequest(**data)
+        data['allocation_id'] = int(data['allocation_id'])
+
+        allocation = Allocation.query.get(data['allocation_id'])
+        if allocation is None:
+            return self.error('No such allocation.')
+        if allocation.group_id not in [
+            g.id for g in self.current_user.accessible_groups
+        ]:
+            return self.error('User does not have access to this allocation.')
+
+        instrument = allocation.instrument
+        if instrument.api_classname is None:
+            return self.error('Instrument has no remote API.')
+
+        if not instrument.api_class.implements_submit():
+            return self.error('Cannot submit followup requests to this Instrument.')
+
+        # validate the payload
+        jsonschema.validate(data['payload'], instrument.api_class.form_json_schema)
```

Comment:
Nice, we need this for the rest of SkyPortal too, as soon as we can figure out how to convert our existing apispec to jsonschema.